src/lib/datepicker/datepicker-input.ts

```diff
@@ -229,14 +228,18 @@ export class MatDatepickerInput<D> implements AfterContentInit, ControlValueAcce
       private _elementRef: ElementRef,
       @Optional() public _dateAdapter: DateAdapter<D>,
       @Optional() @Inject(MAT_DATE_FORMATS) private _dateFormats: MatDateFormats,
-      @Optional() private _formField: MatFormField) {
+      @Optional() private _formField: MatFormField,
+      @Inject(DOCUMENT) document?: any) {
     if (!this._dateAdapter) {
       throw createMissingDateImplError('DateAdapter');
     }
     if (!this._dateFormats) {
       throw createMissingDateImplError('MAT_DATE_FORMATS');
     }
 
+    // @deletion-target 6.0.0 make the `document` a required param.
```

Comment:
nit: move comment above line where param is defined

Reply:
Done.